Usage of URLSearchParams breaks in IE11

octaedro opened the pull request:
Fixes #4497

This PR removes the method `URLSearchParams` (not compatible with IE 11) to use something compatible.
This method was used on the new `Inbox notes` and on the `layout controller`.

### Detailed test instructions:
- Using IE 11.
- Go to `WooCommerce`| `Documentation` | `search-list-control` (URL `/wp-admin/admin.php?page=wc-admin&path=%2Fdevdocs&component=search-list-control`)
- Verify the page loaded normally.
- Press the `Inbox` button that is in the header.
- Verify the error `'URLSearchParams' is undefined ` is not present in the console.

<!--- Note: When displaying information based on sample data, such as SwaggerHub, 
be sure to detail parts affected in Release Notes --->

### Changelog Note:
Fix: Removed URLSearchParams method
<!--- Optional: Enter a changelog note following the WooCommerce core format using prefixes of Enhancement:, Tweak:, Dev:, Fix:, Performance:. If no note is entered, one will be constructed from the title and labels. --->

octaedro committed:
Removed URLSearchParams method

This commit removes the method URLSearchParams to use something compatible with IE

octaedro committed:
Revert "Removed URLSearchParams method"

This reverts commit c2b24b34c93d16264c62b3bf522f4fbc94354d58.

octaedro committed:
Removed URLSearchParams method

This commit removes the method URLSearchParams to use something compatible with IE

timmyc commented on client/layout/controller.js:
```diff
@@ -50,13 +50,24 @@ export const getPages = ( homepageEnabled ) => {
 	const pages = [];
 	const initialBreadcrumbs = [ [ '', wcSettings.woocommerceTranslation ] ];
 
+	const getUrlParams = ( locationSearch ) =>
```
Instead of having this code twice, what do you think about extracting it to a util method and adding some tests around it?

octaedro replied:
It's a great idea @timmyc!
As it will be a `util` used in the header and in the layout components, I created a new folder `client/utils` to keep the utils that are used by different pages of the client.
If you think that it should be different just let me know and I will change it.
Also I added some tests 🙂
The changes are in the commit 9e81c5a

octaedro committed:
Created folder utils with global utils

This commit moves the method "getUrlParams" to a utils file. Also some unit test were added

timmyc reviewed (approved):
Verified the fix in IE11 on browserstack. Thanks for creating the util and adding tests @octaedro
